airflow/models/taskinstance.py

```diff
@@ -1495,8 +1496,9 @@ def _update_ti_state_for_sensing(self, session=NEW_SESSION):
         # Raise exception for sensing state
         raise AirflowSmartSensorException("Task successfully registered in smart sensor.")
 
-    def _execute_task(self, context, task_copy):
+    def _execute_task(self, context, task_orig):
         """Executes Task (optionally with a Timeout) and pushes Xcom results"""
+        task_copy = self.task
```

Comment:
This looks suspect - it isn't a copy anymore but the same object.

(It might make sense in the full diff, but I can't easily see that on the mobile view)

Reply:
No you’re right, this name doesn’t really make much sense. But this is how it’s currently implemented for some reason. In `main`, `_execute_task` is only called once like this:

```python
result = self._execute_task(context, self.task)
```

I left the name alone to minimise diff and definitely wouldn’t object to renaming.